service/gcs/transport/mock.go

```diff
@@ -15,43 +17,35 @@ type MockTransport struct {
 
 // Dial ignores the port, and returns a MockTransport struct.
 func (t *MockTransport) Dial(_ uint32) (Connection, error) {
-	csr, csw := io.Pipe()
-	scr, scw := io.Pipe()
+	fds, err := syscall.Socketpair(syscall.AF_UNIX, syscall.SOCK_STREAM, 0)
```

Comment:
I don't know the expectation here but can this be leaked if any of the calls fail?